pkg/process/dockerproxy/dockerproxy.go

```diff
@@ -0,0 +1,130 @@
+// +build !windows
+
+package dockerproxy
+
+import (
+	"strconv"
+	"strings"
+
+	model "github.com/DataDog/agent-payload/process"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/gopsutil/process"
+)
+
+// Filter keeps track of every docker-proxy instance and filters network traffic going through them
+type Filter struct {
+	proxyByPID    map[int32]*proxy
+	proxyByTarget map[model.Addr]*proxy
+}
+
+type proxy struct {
+	pid    int32
+	ip     string
+	target model.Addr
+}
+
+// NewFilter instantiates a new filter loaded with docker-proxy instance information
+func NewFilter() *Filter {
+	filter := &Filter{
+		proxyByPID:    make(map[int32]*proxy),
+		proxyByTarget: make(map[model.Addr]*proxy),
+	}
+
+	if procs, err := process.AllProcesses(); err == nil {
+		filter.LoadProxies(procs)
+	} else {
+		log.Errorf("error initiating proxy filter: %s", err)
+	}
+
+	return filter
+}
+
+// LoadProxies by inspecting processes information
+func (f *Filter) LoadProxies(procs map[int32]*process.FilledProcess) {
+Loop:
+	for _, p := range procs {
+		if len(p.Cmdline) == 0 {
+			continue
+		}
+
+		if !strings.Contains(p.Cmdline[0], "docker-proxy") {
+			continue
+		}
+
+		proxy := &proxy{pid: p.Pid}
+		// Extract container (proxy target) address
+		for i, arg := range p.Cmdline {
+			if arg == "-container-ip" && i+1 < len(p.Cmdline) {
+				proxy.target.Ip = p.Cmdline[i+1]
+			}
+
+			if arg == "-container-port" && i+1 < len(p.Cmdline) {
+				port, err := strconv.Atoi(p.Cmdline[i+1])
+				if err != nil {
+					continue Loop
+				}
+				proxy.target.Port = int32(port)
+			}
+		}
+
+		log.Debugf("detected docker-proxy with pid=%d target.ip=%s target.port=%d",
+			proxy.pid,
+			proxy.target.Ip,
+			proxy.target.Port,
+		)
+
+		// Add proxy to cache
+		f.proxyByPID[proxy.pid] = proxy
+		f.proxyByTarget[proxy.target] = proxy
+	}
+}
+
+// Filter all connections that have a docker-proxy at one end
+func (f *Filter) Filter(payload *model.Connections) {
+	original := payload.Conns
+	filtered := make([]*model.Connection, 0, len(original))
+
+	for _, c := range original {
```

Comment:
Do we need to have two loops to make sure p.ip is set for every proxy before being able to call `f.isProxied()`?